tasks/appraisal.rake

```diff
@@ -0,0 +1,187 @@
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'install']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+
+    ENV.delete('APPRAISAL_SKIP_BUNDLE_CHECK')
+  end
+
+  desc 'Update Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :update do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'update']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+
+  desc 'Add all platforms to Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :platform do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      next if ruby_version.start_with?('jruby-')
+
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      p lockfile_prefix(ruby_version)
+      Dir["gemfiles/#{lockfile_prefix(ruby_version)}_*.gemfile.lock"].each do |lockfile|
+        gemfile = lockfile.gsub(/\.lock$/, '')
+        cmd << ['env', "BUNDLE_GEMFILE=#{gemfile}",
+                *bundle(ruby_version), 'lock',
+                '--lockfile', lockfile,
+                '--add-platform', 'x86_64-linux']
+        cmd << ['env', "BUNDLE_GEMFILE=#{gemfile}",
+                *bundle(ruby_version), 'lock',
+                '--lockfile', lockfile,
+                '--add-platform', 'aarch64-linux']
+      end
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+end
+
+TRACER_VERSIONS = %w[
+  2.1
+  2.2
+  2.3
+  2.4
+  2.5
+  2.6
+  2.7
+  3.0
+  3.1
+  3.2
+  jruby-9.2.8.0
+  jruby-9.2
+  jruby-9.3
+  jruby-9.4
+  #truffleruby-22.3.0
+].freeze
```

Comment:
Yeah, I guess this comes from the fact that `docker-compose` has `tracer-<version>` services, which I also find a bit awkward.

Tangentially related, I'd like to unify the various rubies naming to be `<RUBY_ENGINE>-<RUBY_ENGINE_VERSION-ish>`. In many cases I found it very annoying to not have the `ruby-` prefix for those `ruby` engine versions.

Reply:
Can you share more about your experience? To be honest, I've been thinking we actually should go in the other direction and drop the minor versions, since it creates a lot of noise just to bump from 3.0.2 to 3.0.3 with no actual changes.

Reply:
ah I was not thinking about the version numbers (hence why I said `VERSION`-ish) but mostly things like adding the `ruby` prefix to C Ruby throughout (`ruby-`, `ruby_`, or whatever)

and then maybe unify stuff like:

- `TRACER_VERSIONS` => `RUBY_VERSIONS`
- `docker-compose --rm tracer-x.y` => `docker-compose --rm ruby-x.y`
- and so on and so forth

Reply:
Ahhhh right. Yeah, I think adding engine and removing minor version from a few random places would be great ;)